salt/config/__init__.py

```diff
@@ -3563,7 +3613,7 @@ def apply_master_config(overrides=None, defaults=None):
         opts['ipc_write_buffer'] = 0
     using_ip_for_id = False
     append_master = False
-    if not opts.get('id'):
+    if not opts.get('id') or opts.get('minion_id_type', 'str') == 'func':
```

Comment:
This is valid to the identical line above. I am not sure this is correct. You are asking to run `get_id()` function if there is no parameter `id` or options set to `func`? Isn't it a bug, because you're supposed to ask it when `minion_id_type` is "str" instead?

And in case you want to `get_id` as usual with `minion_id_type: str` and `id` absent, then it is enough to just do this:
```python
if not opts.get('id', opts.get('minion_id_type' == 'str')):
   ...
```